src/impl/source/profiles/profileActions.ts

```diff
@@ -49,8 +49,13 @@ export default abstract class ProfileActions {
       FileUtils.mkDirByPathSync(profilePath);
     }
 
+    // Query the profiles from org
```

Comment:
Moving this retrieve here force to retrieve all profiles orgs, even if only one profile was passed to the command.

Reply:
@genoud previously it was kept inside profilename iterator, if 5 profile names are passed then this will make 5 API calls.

Also with new refactor, if profilepath -f is passed we are reading all profiles inside the path and passing as profilename list. to improve better performance

Reply:
@manisfdcsfdx  I see. Yes you are right.